### PR TITLE
allow passing absolute INCLUDE_DIRS via catkin_package() into CMake config file in install space

### PR DESCRIPTION
**Pull request by dirk-thomas**
Will need to be backported to _groovy-devel_ branch.

@wjwwood Please review.

### CODE REVIEW

**Comment by hersh** [comment]
Hydro-devel also please?

**Comment by wjwwood** [comment]
Hydro is released from the `groovy-devel` branch, so a back port into `groovy-devel` will make it into hydro :smile:.

**Comment by dirk-thomas** [comment]
If there would be a branch _hydro-devel_ sure - but there isn't :wink: _Groovy_ and _Hydro_ releases are made from the same branch.

**Comment by hersh** [comment]
Excellent.

**Comment by wjwwood** [comment]
+1

**Comment by cottsay** [comment]
This just broke all of my builds for Fedora. The first to break was anything that depends on std-msgs, so I'll quote the error:

```
CMake Error at /opt/ros/hydro/share/std_msgs/cmake/std_msgsConfig.cmake:106 (message):
  Project 'std_msgs' specifies
  '/builddir/build/BUILD/ros-hydro-std-msgs-0.5.8/build/devel/include' as an
  include dir, which is not found.  It does neither exist as an absolute
  directory nor in
  '/opt/ros/hydro//builddir/build/BUILD/ros-hydro-std-msgs-0.5.8/build/devel/include'.
  Ask the maintainer 'Tully Foote <tfoote@osrfoundation.org>' to fix it.
Call Stack (most recent call first):
  /opt/ros/hydro/share/catkin/cmake/catkinConfig.cmake:72 (find_package)
  CMakeLists.txt:14 (find_package)
-- Configuring incomplete, errors occurred!
```

The directory /builddir/build/BUILD/ros-hydro-std-msgs-0.5.8/ was where the package was BUILT, but it was installed to the system root.

tl;dr: The build paths were included in the installed Cmake file, which should not happen.

I'll keep looking into it...

**Comment by cottsay** [comment]
I just saw @dirk-thomas's commits there...looks like that fixed the bug in 0.5.83 that broke the builds. 0.5.84 seems to be working, so I think we're okay.
